### PR TITLE
COVID19 RDT reference app docs, videos, training 

### PR DESCRIPTION
**Pull request by mrjones-plip**
Docs for what is in and how to use the COVID19 RDT reference app for the CHT, per the [grant deliverables](#medic/cht-core/issues/6981).

See #medic/cht-core/issues/6981

### CODE REVIEW

File: content/en/apps/examples/covid-rdt-reference-app.md
Comment (abbyad) [inline content/en/apps/examples/covid-rdt-reference-app.md:122]: Do you have a script that you used internally already for this? It could be useful to post a command or snippet as an example rather than leaving it as an exercise from scratch.

File: content/en/apps/examples/covid-rdt-reference-app.md
Comment (abbyad) [inline content/en/apps/examples/covid-rdt-reference-app.md:127]: It wasn't clear to me what the "Directory structure" header was referring to, or what the filtered file structure criteria was. Here is a suggested adjustment:

```suggestion
## Customizing the reference app

These are the files in the COVID-19 app where you'll want to focus your customization efforts:
```

File: content/en/apps/examples/covid-rdt-reference-app.md
Comment (abbyad) [inline content/en/apps/examples/covid-rdt-reference-app.md:145]: It's nice to include the JSON of reports as reference. Given the "JSON Reference" header though, I was expecting JSON definition with descriptions, rather than a sample -- but that may just be me! Assuming it's not, then it may be nice to clarify the contents of this section in the header.

```suggestion
## Example form submission
```

File: content/en/apps/examples/covid-rdt-reference-app.md
Comment (abbyad) [inline content/en/apps/examples/covid-rdt-reference-app.md:3]: Looking at the examples we have it may make sense to distinguish this as a "testing" workflow

```suggestion
title: "Reference Application using RDT for COVID-19 Testing"
linkTitle: "COVID-19 Testing"
```

File: content/en/apps/examples/covid-rdt-reference-app.md
Comment (abbyad) [inline content/en/apps/examples/covid-rdt-reference-app.md]: Minor adjustment to language, and I think it's clearer to define the acronym without the `s` -- not sure if there is a standard way for that which we should follow though.

```suggestion
Medic has worked with [FIND](https://www.finddx.org) to provide a reference application using rapid diagnostic tests (RDT) for COVID-19 testing with the CHT.  By leveraging the expertise of third-party applications, like [Dimagi's RD-Toolkit](https://github.com/dimagi/rd-toolkit/), you can customize this application in your current or future deployment of the CHT.
```

File: content/en/apps/examples/covid-rdt-reference-app.md
Comment (abbyad) [inline content/en/apps/examples/covid-rdt-reference-app.md:15]: It would be helpful for us at Medic to be less possessive of the CHT ;)

```suggestion
You can find the code for this application in the [CHT Core repository on GitHub](https://github.com/medic/cht-core/tree/master/config/covid-19).
```

File: content/en/apps/examples/covid-rdt-reference-app.md
Comment (abbyad) [inline content/en/apps/examples/covid-rdt-reference-app.md:30]: I wasn't too clear on the original intention of this point. Here is my understanding, feel free to edit further.

```suggestion
 * Best practices for RDT workflows in the CHT
```

File: content/en/apps/examples/covid-rdt-reference-app.md
Comment (abbyad) [inline content/en/apps/examples/covid-rdt-reference-app.md:47]: The diagram uses some notations differently than we do in other diagrams (eg blocks for _parts of_ a form, rather than individual forms). Let's chat directly about some changes that could simplify it.

**Review by abbyad — CHANGES_REQUESTED** [review]
Nice work!! This presents the reference app really well! 

I've made mostly minor suggestions inline, and then ready to go!

**Comment by mrjones-plip** [comment]
yeeessss!  Thanks for the all the feedback here @abbyad! I'll dive into these today.

File: content/en/apps/examples/covid-rdt-reference-app.md
Comment (mrjones-plip) [reply content/en/apps/examples/covid-rdt-reference-app.md:73]: Ah dude - I type `RD-Toolking` or `RD-Timer` or `RD-anthing-but-toolkit` so much! Indeed your fix is the correct one - thank you!

File: content/en/apps/examples/covid-rdt-reference-app.md
Comment (mrjones-plip) [reply content/en/apps/examples/covid-rdt-reference-app.md]: Agreed! And I was trying to do the[ "cursor" in `scrcpy`](https://en.wikipedia.org/wiki/Scrcpy) - but of course you [can turn it on in android natively](https://www.skyneel.com/show-touches-in-android-phone).  I'll look into this!  As well,I'd be happy to do some edits to have "stages" and pause the video to show a interstitial white text on black background "CHW will now select the patient", "RDT test will not be provisioned including a timer and storing the session ID In the CHT" or some such thing.

Otherwise, if there's an in-house CHT demo video expert, I'd be happy to work with them!

File: content/en/apps/examples/covid-rdt-reference-app.md
Comment (mrjones-plip) [reply content/en/apps/examples/covid-rdt-reference-app.md]: Yes! I went back and forth with the `s`, but happy to defer to your preference

File: content/en/apps/examples/covid-rdt-reference-app.md
Comment (mrjones-plip) [reply content/en/apps/examples/covid-rdt-reference-app.md:15]: \o/

File: content/en/apps/examples/covid-rdt-reference-app.md
Comment (mrjones-plip) [reply content/en/apps/examples/covid-rdt-reference-app.md:30]: yeah - this was me repurposing other "example" pages in the docs as templates.  Your change is most welcome!

File: content/en/apps/examples/covid-rdt-reference-app.md
Comment (mrjones-plip) [reply content/en/apps/examples/covid-rdt-reference-app.md:47]: for sure! I knew that i was deviating the well trod-path, but thought it was worth it to denote the different apps, tasks and forms involved.

File: content/en/apps/examples/covid-rdt-reference-app.md
Comment (mrjones-plip) [reply content/en/apps/examples/covid-rdt-reference-app.md]: Good points! Both fixed in 21e13b1

File: content/en/apps/examples/covid-rdt-reference-app.md
Comment (mrjones-plip) [reply content/en/apps/examples/covid-rdt-reference-app.md:135]: This is actually what I did!  The call to the reports API above results in a `outputs.csv` file.  The shell call in this section uses that same file to generate an image from one of the rows.

I don't think it's useful to bulk export the images from a CSV file (say in a while loop in Bash) as likely you'd be doing either an ETL into a 3rd party system, or just raw bulk import w/ no ETL and then on display show the base64 as an image, like we do with the CHT.

Maybe this wasn't clear though and I could add some extra explanation about this? Maybe you disagree and that a `while` script in Bash here is what's needed?

File: content/en/apps/examples/covid-rdt-reference-app.md
Comment (mrjones-plip) [reply content/en/apps/examples/covid-rdt-reference-app.md]: huh - couldn't accept your suggestion - exact same suggestion applied in 5d3f256

**Comment by mrjones-plip** [comment]
@abbyad  - I requested another review to go over what I did based your initial feedback - thanks!!

We still need to tend to:
* redo video w/ cursor, slower/broken up/subtitles
* redo flow/diagram image at top
* pushback on base64 script

File: content/en/apps/examples/covid-rdt-reference-app.md
Comment (abbyad) [reply content/en/apps/examples/covid-rdt-reference-app.md]: Nice! I made a [comment in the commit](https://github.com/medic/cht-docs/commit/21e13b116607c3d6729294afa9d651fd05e742d1#r56798420).

File: content/en/apps/examples/covid-rdt-reference-app.md
Comment (abbyad) [reply content/en/apps/examples/covid-rdt-reference-app.md:135]: Right, not entirely _from scratch_, but from the command to extract a single image. It is likely not obvious from the extraction command what to change to get the other images out for a test, so to make it easier I think including the while loop would be helpful. If you think that there are too many aspects to take into account, then another option would be to mention which part to change to extract images from different lines and then if someone needs to do it programmatically they can figure it out from there.

Also, your comment about ETL is useful, and it could be worth mentioning in the text to give proper context.

Comment (mrjones-plip) [reply content/en/apps/examples/covid-rdt-reference-app.md:141]: Oh - cool cool! thanks for the push-back on the push-back! I've tended to this in b71879f and b24dc48.  Lemme know what ya think!

**Comment by mrjones-plip** [comment]
> Nice to have this as an alert. Can `follow our documentation` be a link to a specific place in our docs? Also, I try to avoid using "our", since I find it to be like "we", which _we_ [try to avoid](https://docs.communityhealthtoolkit.org/contribute/docs/style-guide/#using-we). Perhaps _As per the section on authentication, you need to use a login with `admin` permissions with this stucture:_

So I actually couldn't find docs on requirements to use authentication on our API calls.  I've updated the wording, mostly per your suggestion in 540cbc7.  Lemme know if you have a good link and I'll take your suggestion whole sale!

**Comment by mrjones-plip** [comment]
@abbyad - videos aside, back to you for approval (which I guess it already waiting for your approval so I can't re-send it back to you again?)

File: content/en/apps/examples/covid-rdt-reference-app.md
Comment (abbyad) [inline content/en/apps/examples/covid-rdt-reference-app.md:13]: When rereading this it seems as though it could use a bit more context to understand _why_ you'd want to use third-party applications.

```suggestion
Medic has worked with [FIND](https://www.finddx.org) to build a CHT reference application for COVID-19 point-of-care testing with Rapid Diagnostic Tests (RDT). Using the reference app as an example, CHT app developers can easily include the provisioning and capture of RDT in workflows. These workflows can include third-party applications, like [Dimagi's RD-Toolkit](https://github.com/dimagi/rd-toolkit/), that guide health workers through the use of the RDT.
```

File: content/en/apps/examples/covid-rdt-reference-app.md
Comment (abbyad) [inline content/en/apps/examples/covid-rdt-reference-app.md:6]: Revisiting to align with titles and descriptions in [index page](https://docs.communityhealthtoolkit.org/apps/examples/).

```suggestion
title: "COVID-19 Testing with Rapid Diagnostic Tests"
linkTitle: "COVID-19 Testing"
weight: 
description: >
 CHT example application that uses a third party app to capture the result of a Rapid Diagnostic Test.
```

File: content/en/apps/examples/covid-rdt-reference-app.md
Comment (abbyad) [inline content/en/apps/examples/covid-rdt-reference-app.md:32]: Is cht-android 0.10.0 implied, or should it be stated explicitly?

Comment (abbyad) [reply content/en/apps/examples/covid-rdt-reference-app.md:32]: Having RD-Toolkit is a requirement for this particular app too, where can that be explained?

File: content/en/apps/examples/covid-rdt-reference-app.md
Comment (abbyad) [inline content/en/apps/examples/covid-rdt-reference-app.md:83]:
```suggestion
Like all applications written for the CHT, there are built-in mechanisms to retrieve raw and aggregate data to generate reports and dashboards. Here are some ways that the data can be accessed:
```

File: content/en/apps/examples/covid-rdt-reference-app.md
Comment (abbyad) [inline content/en/apps/examples/covid-rdt-reference-app.md:87]: Seems worthwhile to have the complete name once

```suggestion
 * **[PostgreSQL queries](https://github.com/medic/cht-couch2pg):** CHT ships with a utility to export all the data that the API has to a relational database, Postgres. You have all the raw data the API has, but can now use the power of joins and groupings to come up with totally customizable stats by day, week, month etc. Data can be synched daily or hourly from the CHT. 
```

File: content/en/apps/examples/covid-rdt-reference-app.md
Comment (abbyad) [inline content/en/apps/examples/covid-rdt-reference-app.md:88]: ~We~

```suggestion
 * **Dashboards:** Medic has used both [Klipfolio](https://www.klipfolio.com/) and, more recently, [Superset](https://superset.incubator.apache.org/) to create more complex yet still user-friendly dashboards. This is particular useful for those who need to view the data but wouldn't otherwise be logging in to CHT apps. These dashboards generally access the relational data in the Postgres database as the back end.
```

File: content/en/apps/examples/covid-rdt-reference-app.md
Comment (abbyad) [inline content/en/apps/examples/covid-rdt-reference-app.md:137]: Suggestion for consistency

```suggestion
This code snippet is good just to validate data ad hoc, but more likely you'll be doing either an extract-transform-load (ETL) process into a third-party system, or just import the raw contents and then display the base64 as an image, as done in the CHT.
```

File: content/en/apps/examples/covid-rdt-reference-app.md
Comment (abbyad) [reply content/en/apps/examples/covid-rdt-reference-app.md:141]: Nice! Looks great!!

File: content/en/apps/examples/covid-rdt-reference-app.md
Comment (abbyad) [inline content/en/apps/examples/covid-rdt-reference-app.md:144]: Avoiding the use of "reference" here since it may be more of an "example" app

```suggestion
## Customizing the application
```

**Review by abbyad — CHANGES_REQUESTED** [review]
Overall looks really good! Made some suggestions to try to standardize the language. I don't expect making more suggestions after this set of changes :)

File: content/en/apps/examples/covid-rdt-reference-app.md
Comment (mrjones-plip) [reply content/en/apps/examples/covid-rdt-reference-app.md]: Good questions both! Answered in 78e19ce.

**Comment by mrjones-plip** [comment]
>  I don't expect making more suggestions after this set of changes :)

But then I added a who new paragraph... ;)  No rush! happy to make this right.

File: content/en/apps/examples/covid-rdt-reference-app.md
Comment (abbyad) [inline content/en/apps/examples/covid-rdt-reference-app.md:42]: Helpful new paragraph!

Personal preference to avoid using `utilize`, and have link text match the anchor more closely, along with a clarification:

```suggestion
While this application calls the RD-Toolkit, the integration features in the CHT Core and CHT Android are generic.  This means you could use a different RDT Android application if you prefer. Beyond the scope of RDTs, you could use this integration feature to launch any other Android app to perform an action and save the result in the CHT. To read more about this feature, see the [Android App Launcher section in the Forms reference documentation]({{< ref "apps/reference/forms/app#android-app-launcher" >}}). 
```

**Review by abbyad — CHANGES_REQUESTED** [review]
Nice work, and thanks for pushing through to get this right!

Is the updated diagram included somewhere?

**Comment by mrjones-plip** [comment]
Thanks for the input on the new paragraph @abbyad ! I also forgot to update the new workflow which I just did in f047089

**Review by abbyad — APPROVED** [review]
Awesome work pulling this together and iterating on it!!

**Comment by mrjones-plip** [comment]
Thanks for all the reviewing help @abbyad ! I'll do another micro mini PR with the new video URLs once we publish those.

**Pull request merged**
